### PR TITLE
Mexc createOrder (stopLoss, takeProfit)

### PR DESCRIPTION
**Pull request by Dan-krm**
Related to #12044
Added unified stopPrice to Mexc contract markets.
```
mexc.createSwapOrder (BTC_USDT, 1, 1, 10, 30000, [object Object])
2022-03-16T22:45:09.247Z iteration 0 passed in 877 ms

{
  id: '259206749995677696',
  clientOrderId: undefined,
  timestamp: undefined,    
  datetime: undefined,
  lastTradeTimestamp: undefined,
  status: undefined,
  symbol: 'BTC/USDT',
  type: undefined,
  timeInForce: undefined,
  side: undefined,
  price: undefined,
  stopPrice: undefined,
  average: undefined,
  amount: undefined,
  cost: undefined,
  filled: undefined,
  remaining: undefined,
  fee: undefined,
  trades: [],
  info: { success: true, code: '0', data: '259206749995677696' },
  fees: []
}
```

### CODE REVIEW

**Comment by kroitor** [comment]
@Dan-krm @samgermain @frosty00 @ttodua @carlosmiei @DoctorSlimm @sc0Vu @pcriadoperez 

Let's review this aspect of the `stopPrice` unification to clarify it.
Related issue: https://github.com/ccxt/ccxt/issues/12044
PRs for reference:
- https://github.com/ccxt/ccxt/pull/12302
- https://github.com/ccxt/ccxt/pull/12316

First, there's two types/cases of "stops" and we have to differentiate between the two:

1. A standalone _conditional stop order_ aka _algo order_ – conditional orders can exist by themselves and can be placed separately from other orders.
2. A _stop loss order_ and/or a _take profit stop order_ – those orders can not exist by themselves and are usually tied to another order that opens a margin position (regardless of the market type, whether it is a contract-margin market or a spot-margin market).

This issue https://github.com/ccxt/ccxt/issues/12044 about the `stopPrice` unification is related to type/case (1) **ONLY**.

-------------------------------------------------------------------------------

The usage of type/case (1) is the following:

Suppose you want to place an order (b) if the price meets some condition. To do that, you can place a special order (a) called _"stop conditional"_ order.
A "stop conditional" order (a) is an order to place another "triggered" order (b). That triggered order (b) can be either an order to open a new contract-position, or it could be a spot-order, pretty much any kind of order. It does not care if you have a position or not. When the price condition is met, then the conditional stop order (a) will be triggered, and it will result in placing another triggered order (b) of type/amount/price that the user specifies. So, in general, to place a conditional stop order (a), the user has to supply:

1. conditional `stopPrice` – this is related to the conditional order (a) itself
2. triggered `type` (`'market'` or `'limit'`) – this is related to the order (b) that will be placed by the conditional order, when triggered
3. triggered `amount` – this is related to the order (b) that will be placed by the conditional order, when triggered
4. triggered `price` (if `'limit'`) – this is related to the order (b) that will be placed by the conditional order, when triggered

So, the `stopPrice` conditional order (a) type/case (1) is just a way of conditionally-placing another triggered order (b) if the market meets some condition. It will be triggered when that condition is met. If the desired condition never occurs again, then the conditional order (a) will never be triggered, it will wait until you cancel it. The order (b) won't be placed at all, if the condition is never met.

-------------------------------------------------------------------------------

The usage of type/case (2) is the following:

Suppose you want to open a contract-margin position or spot-margin position by placing an order (a).

You also want to close that position when your desired unrealized profit meets a certain level, in percentage or in absolute prices. At that point you want to realize or _"fix"_ your unrealized profits on that open position. You want to close a profitable position at some point with another order that would fix the profits on that position. That is called _"take profit stop"_ order. (b)

And you also want to close that position if the market goes against your expectation, and when your losses reach a certain level, you want to stop getting further losses, or _"fix" your losses. That is called a _"stop loss"_ order. (c)

In this case, the user wants to place order (a) and attach two stop-orders (b) and (c) to the resulting position that will be opened by order (a).
Therefore, the user needs to specify a set of params for all three orders:

1. opening order (a) amount
2. opening order (a) type (`'market'` or `'limit'`)
3. opening order (a) price (if `'limit'`)
4. closing order (b) `stopLossPrice` – this is optional
5. closing order (c) `takeProfitPrice` – this is optional

Note, that the amount is not specified for orders (b) and (c), cause those orders will usually close the entire `amount` of the position specified in order (a).
Since both orders (b) and (c) are optional, the user can skip any of the two if he wants to close that position manually.
Also, orders of type (b) and (c) can be added/attached later to an already existing open position.

So, stop-loss/take-profit orders cannot exist by themselves, cause their purpose is to fix losses and profits on existing positions (this works for both contract margin and spot margin).

There is no unification schema for type/case (2) at this time. It is yet to be designed. Probably with a pair of extra params like `stopLossPrice` and `takeProfitPrice` (as opposed to `stopPrice` in case (1)).

------------------------------------------------------------------------------

Once again, this issue https://github.com/ccxt/ccxt/issues/12044 about the `stopPrice` unification is related to type/case (1) **ONLY**.

File: js/mexc.js [hunk 1]
```diff
@@ -1801,9 +1801,9 @@ module.exports = class mexc extends Exchange {
             // 'stopLossPrice': this.priceToPrecision (symbol, stopLossPrice),
             // 'takeProfitPrice': this.priceToPrecision (symbol, takeProfitPrice),
         };
-        const stopPrice = this.safeNumber2 (params, 'stopLossPrice', 'stopPrice');
+        const stopPrice = this.safeNumber (params, 'stopLossPrice');
```
Comment (samgermain) [inline js/mexc.js:1804]: `stopPrice` is the unified parameter

Comment (kroitor) [reply js/mexc.js:1804]: @samgermain `stopPrice` is a unified parameter for a different purpose: https://github.com/ccxt/ccxt/pull/12302#issuecomment-1065972543

Comment (samgermain) [reply js/mexc.js:1804]: > @samgermain `stopPrice` is a unified parameter for a different purpose: [#12302 (comment)](https://github.com/ccxt/ccxt/pull/12302#issuecomment-1065972543)

I don't understand the difference between `stopPrice` and `stopLossPrice`

Comment (ttodua) [reply js/mexc.js:1804]: `stopPrice` is specifically for stop-orders (which can be entry order, exit order or sometimes stoploss order too). so, stopLossPrice is specifically for those embeded child-order, which is tied to entry-order (when firing `createOrder`)

**Pull request merged**